common/djangoapps/enrollment/views.py

```diff
@@ -489,3 +490,16 @@ def post(self, request):
                     ).format(username=username, course_id=course_id)
                 }
             )
+        finally:
+            # Assumes that the ecommerce service uses an API key to authenticate.
+            if has_api_key_permissions:
+                current_enrollment = api.get_enrollment(username, unicode(course_id))
```

Comment:
@jimabramson the call is made a second time to get the most current state of the enrollment, after the requested actions have been taken. I've updated the audit log call to handle cases in which the enrollment may remain `None`.